### PR TITLE
[Feature](bangc-ops): rois and offset support nan inf

### PR DESCRIPTION
**Pull request by defei-coder**
Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

rois and offset support nan\inf.

## 2. Modification

- bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu

## 3. Test Report

[----------] Global test environment tear-down
[ SUMMARY  ] Total 70 cases of 2 op(s).
ALL PASSED.
[==========] 70 test cases from 2 test suites ran. (19188 ms total)
[  PASSED  ] 70 test cases.

### 3.4 Summary Analysis

rois 支持 nan，rois 不支持 inf，当 roi 坐标为 inf 时，将产生 0~2147483647 的 for 循环；offset 支持 nan or inf.

### CODE REVIEW

File: bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu [hunk 1]
```diff
@@ -46,8 +57,8 @@ __mlu_func__ void bilinearInterpolate(const int input_width, T y, T x, T *w1,
   }
 
   if (x <= 0) x = 0;
-
-  *x_low = int(x);
+  bool nan_flag = std::isnan(x);
+  *x_low = nan_flag ? 0 : int(x);
```
Comment (mahxn0) [inline bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu:61]: mlu kernel里面int(nan)就是0,这里不需要用std::isnan(x)去单独判断了．

File: bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu [hunk 2]
```diff
@@ -159,7 +178,8 @@ __mlu_func__ void MLUMultiKernelDeformRoiPoolForward(
           y = 0;
         }
         int y_low = 0, y_high = 0;
-        y_low = int(y);
+        bool nan_flag = std::isnan(y);
+        y_low = nan_flag ? 0 : int(y);
```
Comment (mahxn0) [inline bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu:182]: 同上

File: bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu [hunk 3]
```diff
@@ -228,7 +248,8 @@ __mlu_func__ void MLUMultiKernelDeformRoiPoolForward(
                 if (y <= 0) {
                   y = 0;
                 }
-                y_low = int(y);
+                nan_flag = std::isnan(y);
```
Comment (mahxn0) [inline bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu:251]: 同上

File: bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu [hunk 4]
```diff
@@ -428,7 +455,8 @@ __mlu_func__ void MLUMultiKernelDeformRoiPoolBackward(
           y_tmp = 0;
         }
         int y_low = 0, y_high = 0;
-        y_low = int(y_tmp);
+        bool nan_flag = std::isnan(y_tmp);
+        y_low = nan_flag ? 0 : int(y_tmp);
```
Comment (mahxn0) [inline bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu:459]: 同上

File: bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu [hunk 5]
```diff
@@ -530,7 +558,8 @@ __mlu_func__ void MLUMultiKernelDeformRoiPoolBackward(
                   if (y_tmp <= 0) {
                     y_tmp = 0;
                   }
-                  y_low_tmp = int(y_tmp);
+                  nan_flag = std::isnan(y_tmp);
+                  y_low_tmp = nan_flag ? 0 : int(y_tmp);
```
Comment (mahxn0) [inline bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu:562]: 同上

File: bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu (hunk 1 above)
Comment (mahxn0) [reply bangc-ops/kernels/deform_roi_pool/deform_roi_pool_union1.mlu:64]: 更正：mlu370是int_max，mlu590是0

**Review by mahxn0 — APPROVED** [review]

**Review by Unireverse — APPROVED** [review]

**Review by Wickyzheng — APPROVED** [review]

**Pull request merged**